contracts/core/SoundEditionV1a.sol

```diff
@@ -0,0 +1,92 @@
+ ███████████████████████████████████████████████████████████
+ ███████████████████████████████████████████████████████████▒
+                              ███████████████████████████████████████████████████████████▒
+                              ▓██████████████████████████████████████████████████████████▒
+                               ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓███████████████████████████████▒
+ █████████████████████████████                             ▒█████████████████████████████▒
+ ██████████████████████████████                            ▒█████████████████████████████▒
+ ██████████████████████████████▓▒▒▒▒▒▒▒▒▒▒▒▒▒              ▒█████████████████████████████▒
+ ████████████████████████████████████████████▒             ▒█████████████████████████████▒
+ ████████████████████████████████████████████▒             ▒█████████████████████████████▒
+ ▒▒▒▒▒▒▒▒▒▒▒▒▒▒███████████████████████████████▓▓▓▓▓▓▓▓▓▓▓▓▓███████████████▓▒▒▒▒▒▒▒▒▒▒▒▒▒▒
+               ▓██████████████████████████████████████████████████████████▒
+               ▓██████████████████████████████████████████████████████████
+*/
+
+import { ISoundEditionV1a } from "./interfaces/ISoundEditionV1a.sol";
+import { ISoundEditionV1 } from "./interfaces/ISoundEditionV1.sol";
+import { SoundEditionV1 } from "./SoundEditionV1.sol";
+
+import { IMetadataModuleTrigger } from "./interfaces/IMetadataModuleTrigger.sol";
+
+/**
+ * @title SoundEditionV1a
+ * @notice The Sound Edition contract fork with premint metadata module trigger
+ */
+contract SoundEditionV1a is ISoundEditionV1a, SoundEditionV1 {
+    // =============================================================
+    //                           CONSTANTS
+    // =============================================================
+
+    /**
+     * @dev The boolean flag on whether the `mintRandomness` is enabled.
+     */
+    uint8 public constant METADATA_TRIGGER_ENABLED_FLAG = 1 << 2;
+
+    // =============================================================
+    //               PUBLIC / EXTERNAL VIEW FUNCTIONS
+    // =============================================================
+
+    /**
+     * @inheritdoc ISoundEditionV1a
+     */
+    function metadataTriggerEnabled() public view returns (bool) {
+        return _flags & METADATA_TRIGGER_ENABLED_FLAG != 0;
+    }
+
+    /**
+     * @inheritdoc SoundEditionV1
+     */
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        override(SoundEditionV1, ISoundEditionV1)
+        returns (bool)
+    {
+        return interfaceId == type(ISoundEditionV1a).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    // =============================================================
+    //                  INTERNAL / PRIVATE HELPERS
+    // =============================================================
+    /**
+     * @dev Triggers the metadata module
+     */
+    modifier triggersMetadataUpdate(uint256 quantity) {
```

Comment:
The edition contract should be as minimal and unopinionated as possible. Can this logic be moved into a minter?

Reply:
Yes, but I liked having the ability to use all the standard sound minters without modification. I think you’re right that the logic makes more sense to move it there though. Will do!